nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -424,6 +440,35 @@ void DockingServer::doInitialPerception(Dock * dock, geometry_msgs::msg::PoseSta
   }
 }
 
+void DockingServer::rotateToDock(const geometry_msgs::msg::PoseStamped & dock_pose)
```

Comment:
Design question: Can we add this to the controller? This feels like a controller action perhaps (but open to keep here if you think it should be)